src/jbake/content/2022/08/micronaut-test-resources.adoc

```diff
@@ -0,0 +1,568 @@
+        }
+        return Optional.empty();
+    }
+
+}
+----
+<1> Tells that this resolver can resolve the `my.user.name` property
+<2> Returns the value of the `my.user.name` property
+
+And in order for the resolver to be discovered, we need to create the `src/testResources/resources/META-INF/services/io.micronaut.testresources.core.TestResourcesResolver` file with the following contents:
+
+[source]
+----
+demo.MyTestResource
+----
+
+Now let's write a test for this by adding the `src/test/java/demo/GreeterTest.java` file:
+
+[source,java]
+----
+package demo;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@MicronautTest
+class GreeterTest {
+
+    @Inject
+    Greeter greeter;
+
+
+    @Test
+    @DisplayName("Says hello")
+    void saysHello() {
+        greeter.sayHello();
+        Assertions.assertEquals("Hello, world!", greeter.getGreeting());
+    }
+
+}
+----
+
+Now if you run `./gradlew test`, you will notice that Gradle will compile your custom test resource resolver, and when the test starts, you will read the following line:
+
+`Loaded 3 test resources resolvers: demo.MyTestResource, io.micronaut.testresources.mysql.MySQLTestResourceProvider, io.micronaut.testresources.testcontainers.GenericTestContainerProvider`
+
+So when the `Greeter` bean is created, it will read the value of the `my.user.name` property by calling your custom test resolver!
+Of course this is a very simple example, and I recommend that you take a look at the https://github.com/micronaut-projects/micronaut-test-resources[Micronaut Test Resources sources] for more examples of implementing resolvers.
+
+== Conclusion
+
+In this blog post, we've explored the new {mtr}[Micronaut Test Resources] module, which will greatly simplify development of Micronaut applications which depend on external services like databases or messaging queues.
+It works by _simplifying_ configuration, by removing lines which used to be present, but now are dynamically resolved, like `datasources.default.url`.
+Test resources are handled in a separate process, the test resources server, which is responsible for handling their lifecycle.
+This also makes it possible to share the resources (containers, databases, ...) between independent builds.
+For advanced use cases, {mtr}[Micronaut Test Resources] provides _configuration based_ resources creation.
+
+Last but not least, {mtr}[Micronaut Test Resources] is an extensible framework which will let you implement your own test resources in case the built-in ones miss a feature.
+
+Special thanks to twitter.com/tim_yates/[Tim Yates] for his hard work on upgrading the Micronaut Guides to use test resources, and https://twitter.com/alvaro_sanchez[Álvaro Sanchez-Mariscal] for his support on the Maven plugin!
```

Comment:
Runs in late...

Missing `https://` to make me a link. Thanks for the mention btw 😎

Reply:
Fixed, thanks!